lib/cc/engine/analyzers/file_list.rb

```diff
@@ -30,21 +30,7 @@ def paths
         end
 
         def engine_paths
-          if current_language.is_a?(Hash)
-            current_language["paths"]
-          end
-        end
-
-        def current_language
-          if engine_languages.is_a?(Hash)
-            @current_language ||= engine_languages.fetch(language, {})
-          end
-        end
-
-        def engine_languages
-          @engine_language ||= engine_config.
-            fetch("config", {}).
-            fetch("languages", {})
+          @engine_config.language(language).fetch("paths", nil)
```

Comment:
instead of other areas of the codebase being concerned with the structure of this data in `engine_config`, what are your thoughts on creating public methods to get at this data so `EngineConfig` is solely responsible for all of its data?